src/org/centos/pipeline/PipelineUtils.groovy

```diff
@@ -1060,3 +1060,34 @@ def watchForMessages(String msg_provider, String message) {
         }
     }
 }
+
+/**
+ * Figures out current OpenShift project name. That is the project the Jenkins is running in.
+ *
+ * The method assumes that the Jenkins instance has kubernetes plugin installed and properly configured.
+ */
+def getDefaultOpenShiftProject() {
+    return openshift.withCluster() {
+        openshift.project()
+    }
+}
+
+/**
+ * Figures out the Docker registry URL which is supposed to host all the images for current OpenShift project.
+ *
+ * The method uses the 'jenkins' image and assumes that it is part of the current project and that other images are
+ * in the same project as the 'jenkins' image. This is not 100% bullet proof, but should be good enough as starting point.
+ *
+ */
+def getOpenShiftDockerRegistryUrl() {
+    return openshift.withCluster() {
+        def jenkinsImageUrl = openshift.raw("get is jenkins --template='{{.status.dockerImageRepository}}'").out
```

Comment:
This is way too fragile, as different projects can have different images names. I'll try to figure out  better way to get the image registry.

Reply:
Slightly different solution (not relying on the presence of any specific container):
1) Create dummy imagesteam: `oc create imagestream dummy`
2) Get the registry URL from the `dummy` imagestream: `oc get imagestream dummy -o=jsonpath="{.status.dockerImageRepository}"` + parse the URL (same as above)
3) Remove the imagestream `dummy`.

This is definitely less error prone, but it requires two additional API calls every time. I am not sure if this can be an issue or not though. Any comments more than welcome.

Reply:
I am not crazy about creating/removing a dummy imagestream.
I did find this: https://github.com/openshift/jenkins/commit/68d730e67597970f790d5660257869ceb1f67d37#diff-8527116e66492691f95b5090ad6d719e

- you can get all the names of imagestreams.
- you loop over each name until you find what you want based on your split method check to record your docker reg URL.
- once you find one, you exit your loop.

Reply:
I am not sure I understand -- how do I decide which imagestream name is the correct one? The URL will likely have the same structure, being it the internal docker registry, or any other (external) registry. Without knowing e.g. which imagestream I should look at, I don't see way to get the correct URL.

Reply:
I guess I have never seen an Openshift tenant that had imagestreams that were not local...

Reply:
We were using that extensively in middleware. The images were built elsewhere and our imagestreams pointed to external registry. But that's of course not the case here, so we should be fine.

Reply:
Then we should be careful then...If we go the create/delete routem, we would need to ensure the dummy is a unique name...

Reply:
Don't get me wrong -- I don't like the create/delete route :)

After thinking about this more, I would actually just leave it as is for now as it works. If we encounter issues in future then we can perhaps fallback to the create/delete option.